### PR TITLE
Optimized Distance Calculations

### PR DESCRIPTION
**Pull request by joetex**
## Description

I changed all distance calculations to used only the squared version, without using Math.sqrt.  I updated the related expressions to use squared values to compensate.
## Motivation and Context

Using Math.sqrt or variations is extremely expensive.  This was
simplified using the distance squared, rather than the sqrt of distance.

When using the squared distance, you need to ensure any related
expressions are also squared.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bot.user.js [hunk 1]
```diff
@@ -480,7 +453,7 @@ var bot = (function() {
                 for (var j = 0; j < nIter; ++j) {
                     var p2 = sortedFood[j];
                     var dist = canvas.getDistance(p1.xx, p1.yy, p2.xx, p2.yy);
-                    if (dist < 150) {
+                    if (dist < 22500) {
```
Comment (ChadSki) [inline bot.user.js:456]: Magic cutoff number? How did you figure out what was a good number?

**Comment by ChadSki** [comment]
> +24 −51

I love net-deletion commits!

File: bot.user.js [hunk 2]
```diff
@@ -259,26 +238,20 @@ var canvas = (function() {
...
-                        canvas.drawDot(collisionPointX, collisionPointY, circle2.radius, 'cyan', true);
-                        canvas.drawDot(circle2.x, circle2.y, circle2.radius, 'red', true);
-                    }
-                    return true;
-                }
-            }
+			var bothRadiiSqr = bothRadii*bothRadii;
+			var xDist = circle1.x - circle2.x;
+			var yDist = circle1.y - circle2.y;
+			var distance = xDist*xDist + yDist*yDist;
```
Comment (ChadSki) [inline bot.user.js]: At first I was going to say "don't we have a function for this?"

But now that I see [our function](https://github.com/ErmiyaEskandary/Slither.io-bot/blob/master/bot.user.js#L213-L245), maybe your way is better :worried:

File: bot.user.js [hunk 3]
```diff
@@ -214,34 +219,8 @@ var canvas = (function() {
...
-                }
-                cSquare--;
-                diff = num - (cSquare * cSquare);
-                return cSquare + (diff / (cSquare * 2));
-            };
-            var distance = appSqrt((xDistance * xDistance) + (yDistance * yDistance));
-           
-            return distance;
+			
+			return xDistance*xDistance + yDistance*yDistance;
```
Comment (ChadSki) [inline bot.user.js:223]: Wait, you're returning the distance squared, not the real distance. For the function name to remain accurate you would have to take the square root of that return value.

File: bot.user.js [hunk 4]
```diff
@@ -259,26 +238,20 @@ var canvas = (function() {
...
-                    }
-                    return true;
-                }
-            }
+			var bothRadiiSqr = bothRadii*bothRadii;
+			var xDist = circle1.x - circle2.x;
+			var yDist = circle1.y - circle2.y;
+			var distance = xDist*xDist + yDist*yDist;
+			
+			if (distance < bothRadiiSqr) {
```
Comment (ChadSki) [inline bot.user.js:246]: Clever approach to compare the squares, so that you can skip a square root calculation. :+1:

**Comment by ermiyaeskandary** [comment]
@OneEyed Please sign the CLA linked in the README.

**Comment by ermiyaeskandary** [comment]
Thanks

**Comment by ermiyaeskandary** [comment]
Approved Anyone else approving @chancity @ChadSki ?

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/130/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Pull request merged**